ssv/spectest/tests/runner/postconsensus/pre_decided.go

```diff
@@ -115,4 +81,56 @@ func PreDecided() tests.SpecTest {
 			},
 		},
 	}
+
+	// proposerV creates a test specification for versioned proposer.
+	proposerV := func(version spec.DataVersion) *tests.MsgProcessingSpecTest {
+		return &tests.MsgProcessingSpecTest{
+			Name:   fmt.Sprintf("proposer (%s)", version.String()),
+			Runner: testingutils.ProposerRunner(ks),
+			Duty:   testingutils.TestingProposerDutyV(version),
+			Messages: []*types.SSVMessage{
+				testingutils.SSVMsgProposer(nil, testingutils.PreConsensusRandaoDifferentSignerMsgV(ks.Shares[1], ks.Shares[1], 1, 1, version)),
+				testingutils.SSVMsgProposer(nil, testingutils.PreConsensusRandaoDifferentSignerMsgV(ks.Shares[2], ks.Shares[2], 2, 2, version)),
+				testingutils.SSVMsgProposer(nil, testingutils.PreConsensusRandaoDifferentSignerMsgV(ks.Shares[3], ks.Shares[3], 3, 3, version)),
+
+				testingutils.SSVMsgProposer(nil, testingutils.PostConsensusProposerMsgV(ks.Shares[1], 1, version)),
+			},
+			PostDutyRunnerStateRoot: preDecidedProposerSC(version).Root(),
+			PostDutyRunnerState:     preDecidedProposerSC(version).ExpectedState,
+			OutputMessages: []*types.SignedPartialSignatureMessage{
+				testingutils.PreConsensusRandaoMsgV(ks.Shares[1], 1, version),
+			},
+			BeaconBroadcastedRoots: []string{},
+			ExpectedError:          err,
```

Comment:
@alonmuroch or @lior-blox
I don't get something...
In this case shouldn't this be a "future msg" here?

What happens if a peer decides before me and then sends a post consensus message... why should it be an error?

Reply:
This is not a qbft msg so not a "future msg". 
It can be so that's why we should have a smart msg queue in the implementation

Reply:
Currently I understood from @lior-blox that we don't have a msq queue in impl.
He also convinced me that it is actually fine the protocol works this way and it can be considered as an optimization.